pkg/liqo-controller-manager/namespaceMap-controller/ensure_remote_namespaces.go

```diff
@@ -0,0 +1,203 @@
+				Phase:           mapsv1alpha1.MappingCreationLoopBackOff,
+			}
+			errorCondition = 1
+			klog.Errorf("%s -> Namespace '%s' cannot be created inside cluster '%s'", err,
+				localName, nm.Labels[liqoconst.RemoteClusterID])
+			continue
+		}
+		nm.Status.CurrentMapping[localName] = mapsv1alpha1.RemoteNamespaceStatus{
+			RemoteNamespace: remoteName,
+			Phase:           mapsv1alpha1.MappingAccepted,
+		}
+	}
+	return errorCondition
+}
+
+// This function deletes a remote Namespace from the remote cluster, the right client to use is chosen
+// by NamespaceMap's cluster-id. This function return nil (success) only when the remote Namespace is really deleted,
+// so the get Api return a "NotFound".
+func (r *NamespaceMapReconciler) deleteRemoteNamespace(clusterID, remoteName string) error {
+	if err := r.checkRemoteClientPresence(clusterID); err != nil {
+		return err
+	}
+	if err := r.checkLocalClusterID(); err != nil {
+		return err
+	}
+
+	remoteNamespace := &corev1.Namespace{}
+	if err := r.RemoteClients[clusterID].Get(context.TODO(), types.NamespacedName{Name: remoteName}, remoteNamespace); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("The namespace '%s' is correctly deleted from the remote cluster: '%s'", remoteName, clusterID)
+			return nil
+		}
+		klog.Errorf("unable to get the remote namespace '%s'", remoteName)
+		return err
+	}
+
+	// Check if it is a namespace created by the NamespaceMap controller.
+	annotationKey := fmt.Sprintf("%s-%s/%s", remoteNamespaceAnnotationPrefix, r.LocalClusterID, liqoSuffix)
+	if value, ok := remoteNamespace.Annotations[annotationKey]; !ok || value != remoteNamespaceAnnotationValue {
+		klog.Infof("No remote namespaces with name '%s' created through Liqo mechanism inside the remote cluster: '%s'",
+			remoteName, clusterID)
+		return nil
+	}
+
+	if remoteNamespace.DeletionTimestamp.IsZero() {
+		if err := r.RemoteClients[clusterID].Delete(context.TODO(), remoteNamespace); err != nil {
+			klog.Errorf("unable to delete the namespace '%s' from the remote cluster: '%s'", remoteName, clusterID)
+			return err
+		}
+		klog.Infof("The deletion timestamp is correctly set on the namespace '%s'", remoteName)
+	}
+	return fmt.Errorf("remote Namespace '%s' terminating", remoteName)
+}
+
+// This function checks if remote namespaces requested to be deleted are really deleted.
+func (r *NamespaceMapReconciler) ensureRemoteNamespacesDeletion(nm *mapsv1alpha1.NamespaceMap) int {
+	errorCondition := 0
+	for localName, remoteStatus := range nm.Status.CurrentMapping {
+		if _, ok := nm.Spec.DesiredMapping[localName]; ok && nm.DeletionTimestamp.IsZero() {
+			continue
+		}
+		if err := r.deleteRemoteNamespace(nm.Labels[liqoconst.RemoteClusterID], remoteStatus.RemoteNamespace); err != nil {
+			nm.Status.CurrentMapping[localName] = mapsv1alpha1.RemoteNamespaceStatus{
+				RemoteNamespace: remoteStatus.RemoteNamespace,
+				Phase:           mapsv1alpha1.MappingTerminating,
+			}
+			errorCondition = 1
```

Comment:
If you want to return a counter this should be incremented instead of set to one, otherwise use a boolean variable.

Reply:
same as before